cunumeric/array.py

```diff
@@ -3805,3 +3835,69 @@ def _perform_where(cls, mask, one, two) -> ndarray:
         out = ndarray(shape=out_shape, dtype=common_type, inputs=args)
         out._thunk.where(mask, one, two)
         return out
+
+    @classmethod
+    def _perform_scan(
+        cls, op, src, axis=None, dtype=None, out=None, nan0=False
+    ) -> ndarray:
+        if dtype is None:
+            if out is None:
+                if np.issubdtype(src.dtype, np.integer):
+                    # Set dtype to default platform integer
+                    dtype = np.int_
+                else:
+                    dtype = src.dtype
+            else:
+                dtype = out.dtype
+        if (
+            np.issubdtype(src.dtype, np.floating) or np.iscomplexobj(src)
```

Comment:
this could be just `src.dtype.kind in ("f", "c")`

Reply:
90427886a52de634b883d2a0590ed15b65368064